### DISCUSSION TITLE
Audio Only - is it possible to not save the metadata?

### DISCUSSION THREAD

**Discussion by QuinGold** [aclap-dev/video-downloadhelper#1330] [Help and Q&A]
### Your question

I am downloading as "Audio Only" music that accompanies sheet music so I can have it in my Apple Music Library rather than having to go to the website to play it. The files downloaded as Audio Only are .mp3 files and the Title, Album & Artist names are included in metadata. I then change the filename from a generic "piano1.mp3" etc. to a specific filename with Composer-Title-Opus etc. But when I import the folder of files into Apple Music on Mac, it uses the embedded Title from the metadata which isn't as useful as my filename, and sometimes the Titles are just so vague as to be useless (like "Bach 2" etc.)

My question: Is it possible to save an Audio Only file without the metadata? I haven't found an easy way to strip it out. 

My workaround: I open the .mp3 files in QuickTime, then File > Export > Audio Only. This saves the file again as an .m4a file and strips out the metadata. Now Apple Music uses my filename as the Title. Then I Get Info on all the songs and apply the Album name and Artist again. 

If there is an easier way on Mac, let me know. I would love to cut out some steps as I have hundreds of songs. Thanks! 

### Addon & CoApp version and details

version: 9.1.0.41
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"07af69178bff4630"}
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "/Users/xxx/dwhelper",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive"
        }
    },
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": false,
    "open_count_store": 122,
    "successfull_dl": 45
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

VDH does not offer any way of dealing with such details.  VDH just downloads whatever the web site serves up.  If that includes metadata, then VDH faithfully preserves it all.

Strictly speaking, this is now about to go off topic.

I know less than diddly squat about Mac.

I do know that with ffmpeg you can set any metadata you like on a file.  You use an output parameter called `-metadata` & that will set any item of metadata to whatever value you provide.  For example:

-metadata title="Whatever you want"

To remove an item of metadata, you would code something like this:

-metadata title=""

You would code a series of `-metadata` parameters, each specifying a different attribute of metadata to write into the output file.  The general syntax of ffmpeg is this:

ffmpeg globalparameters inputparameters -i "Input file name" outputparameters "Output file name" redirection

The `redirection` thing I'm showing at the end is specific to Windows.  In order to capture the output of ffmpeg in a log file, you would code this for `redirection`:

2>"Log file name"

In my diddly squat ignorance of Mac, I can't tell you what the analogous thing is for you.  I would assume, though, that you would know.  If you don't code something like this, the execution of ffmpeg just flashes messages by in the command window & you ultimately will have no idea what it did nor whether it was successful.  So capturing the output is essential to not only verify what it did but help you in debugging.  You should expect to have to code ffmpeg invocations over & over until you hit upon a combination that does what you want.  Obviously, you will not be typing this command in a command window each time.  You will be writing a .bat file.  That would be more Windows-speak.  You need to write a command script for all of this in whatever format Mac requires.

You already have ffmpeg.  It is part of the CoApp package.  If Mac is anything like Windows, the VDH settings tell you where that is located:

![#01](https://github.com/user-attachments/assets/8345c155-12dc-4031-a67b-854510f44257)

The approach I would take is to supply a series of `-metadata` parameters with attributes being set to values of `""`.  I would then code the `"Output file name"` to include all the things like artist, track name, album, whatever it is your personal naming convention might be.

In this scenario, I would be sure to include this among your `outputparameters` along with your `-metadata` parameters:

-codec: copy

Without that, everything would still work, but ffmpeg would do a whole lot more processing, burning up your CPU as it goes.  It might also make each invocation of ffmpeg last rather a long time, even on a powerful CPU.

I would also include `-y` among the `outputparameters`.  That tells ffmpeg to overwrite the output file if it already exists.  Without this parameter, ffmpeg pauses & waits for you to authorize the overwrite.  Since you should expect to be getting to a good solution via trial & error, you will want to code `-y`.

Failing that, I suppose what you are already doing is equivalent to what I'm suggesting.  Perhaps using ffmpeg would do everything you want in a single step, but you still have to pretty much hand-craft an ffmpeg command for each file you want to process.  I'm not sure using ffmpeg is any better than what you're doing now.

**Reply by QuinGold** [reply]
Thank you for your detailed response. I'll certainly look into ffmpeg. I'd be happy if I could leave the title blank and keep the same filename. I'm also checking out Audacity as it seems you can batch export multiple files as that's really what is slowing me down.

**Reply by QuinGold** [reply]
Update: I gave up on Audacity as it seem to resample the file when it exports. QT Export as Audio seems the best solution right now.
